docs/data/base/components/input/OTPInput.js

```diff
@@ -0,0 +1,204 @@
+import * as React from 'react';
+import { Input as BaseInput } from '@mui/base/Input';
+import { Box, styled } from '@mui/system';
+
+function OTP({ seperator, inputCount, otp, setOtp }) {
+  const inputRefs = React.useRef(new Array(inputCount).fill(null));
+
+  const focusInput = (targetIndex) => {
+    const targetInput = inputRefs.current[targetIndex];
+    targetInput.focus();
+  };
+
+  const selectInput = (targetIndex) => {
+    const targetInput = inputRefs.current[targetIndex];
+    targetInput.select();
+  };
+
+  const handleKeyDown = (event, currentIndex) => {
+    switch (event.key) {
+      case 'ArrowLeft':
+        event.preventDefault();
+        if (currentIndex > 0) {
+          focusInput(currentIndex - 1);
+          selectInput(currentIndex - 1);
+        }
+        break;
+      case 'ArrowRight':
+        event.preventDefault();
+        if (currentIndex < inputCount - 1) {
+          focusInput(currentIndex + 1);
+          selectInput(currentIndex + 1);
+        }
+        break;
+      case 'Backspace':
+        event.preventDefault();
+        if (currentIndex > 0) {
+          focusInput(currentIndex - 1);
+          selectInput(currentIndex - 1);
+        }
+
+        setOtp((prev) => {
+          const otpArray = [...prev];
+          otpArray.splice(currentIndex, 1);
+          otpArray.push('');
+          return otpArray;
+        });
+
+        break;
+      default:
+        break;
+    }
+  };
+
+  const handleChange = (event, currentIndex) => {
+    const value = event.target.value;
+    let indexToEnter = 0;
+
+    while (indexToEnter <= currentIndex) {
+      if (otp[indexToEnter] && indexToEnter < currentIndex) {
+        indexToEnter += 1;
+      } else {
+        break;
+      }
+    }
+    setOtp((prev) => {
+      const otpArray = [...prev];
+      const lastValue = value[value.length - 1];
+      otpArray[indexToEnter] = lastValue;
+      return otpArray;
+    });
+    if (value !== '') {
+      if (currentIndex < inputCount - 1) {
+        focusInput(currentIndex + 1);
+      }
+    }
+  };
+
+  const handleClick = (event, currentIndex) => {
+    selectInput(currentIndex);
+  };
+
+  const handlePaste = (event, currentIndex) => {
```

Comment:
This works great! I'm wondering if pasting in any of the digits shouldn't replace the whole value, though (so if your value is [9][9][9][9][9], have a cursor in the third input and have 12345 in the clipboard, the component should display [1][2][3][4][5] instead of [9][9][1][2][3]), but I'm not 100% sure about that.

Reply:
Both mentioned benchmarks displays [9][9][1][2][3] on pasting. so i reflected same behaviour.